plugins/canideploy/app/models/canideploy.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+require 'jenkins_api_client'
+require 'singleton'
+# checks jenkins staging status and publishes unstable builds checklist
+class Canideploy
+  include Singleton
+
+  URL = ENV['JENKINS_URL'] # 'https://jenkins.zendesk.com'
+  USERNAME = ENV['JENKINS_USERNAME']
+  API_KEY = ENV['JENKINS_API_KEY']
+  JENKINS_CHECKLIST = ENV['JENKINS_STAGING_VIEW'] # '/view/StagingStatus'
+
+  def initialize
+    @client ||= \
+      JenkinsApi::Client.new(
+        server_url: URL,
+        username: USERNAME,
+        password: API_KEY
+      )
+  end
+
+  def checklist
```

Comment:
some pass/fail boolean would be nice so we can not show this when it passes (reduce clutter)